service/controller/resource/deployment/worker_deployment.go

```diff
@@ -383,9 +389,19 @@ func newWorkerDeployments(customResource v1alpha1.KVMConfig, release *releasev1a
 			},
 		}
 		addCoreComponentsAnnotations(deployment, release)
+		// in case of adding additional env vars
+		addConditionalEnvVarsToK8SKVMContainer(deployment, []corev1.EnvVar{key.HostVolumesToEnvVar(capabilities.HostVolumes)})
 
 		deployments = append(deployments, deployment)
 	}
 
 	return deployments, nil
 }
+
+func addConditionalEnvVarsToK8SKVMContainer(deployment *v1.Deployment, envVars []corev1.EnvVar) {
+	for _, container := range deployment.Spec.Template.Spec.Containers {
+		if container.Name == key.K8SKVMContainerName {
+			container.Env = append(container.Env, envVars...)
+		}
+	}
```

Comment:
`container` is a value rather than a reference in the loop so I think you'll need to do something like this to get it to change the passed in deployment:
```suggestion
	for i, container := range deployment.Spec.Template.Spec.Containers {
		if container.Name == key.K8SKVMContainerName {
			container.Env = append(container.Env, envVars...)
			deployment.Spec.Template.Spec.Containers[i] = container
		}
	}
```

Reply:
good point. if the slice is passed as a reference , all the nested slices will be just value right?